src/main/kotlin/io/github/gradlenexus/publishplugin/NexusPublishPlugin.kt

```diff
@@ -75,80 +79,98 @@ class NexusPublishPlugin : Plugin<Project> {
-            password.convention(rootProject.provider { rootProject.findProperty("${name}Password") as? String })
-            publicationType.convention(PublicationType.MAVEN)
-
             val repository = this
-            val retrieveStagingProfileTask = rootProject.tasks.register<RetrieveStagingProfile>("retrieve${capitalizedName}StagingProfile", rootProject.objects, extension, repository)
+
+            repository.username.convention(rootProject.provider { rootProject.findProperty("${name}Username") as? String })
+            repository.password.convention(rootProject.provider { rootProject.findProperty("${name}Password") as? String })
+            repository.publicationType.convention(PublicationType.MAVEN)
+
+            @Suppress("UNUSED_VARIABLE") // Keep it consistent.
+            val retrieveStagingProfileTask = rootProject.tasks.register<RetrieveStagingProfile>(
+                "retrieve${capitalizedName}StagingProfile"
+            ) {
+                this.group = PublishingPlugin.PUBLISH_TASK_GROUP
+                this.description = "Gets and displays a staging profile id for a given repository and package group. " +
+                    "This is a diagnostic task to get the value and " +
+                    "put it into the NexusRepository configuration closure as stagingProfileId."
+                this.repository.convention(repository)
+                this.packageGroup.convention(extension.packageGroup)
+            }
             val initializeTask = rootProject.tasks.register<InitializeNexusStagingRepository>(
                 "initialize${capitalizedName}StagingRepository",
-                rootProject.objects,
-                extension,
-                repository,
                 registry
             )
+            initializeTask {
+                this.group = PublishingPlugin.PUBLISH_TASK_GROUP
+                this.description = "Initializes the staging repository in '${repository.name}' Nexus instance."
+                this.repository.convention(repository)
+                this.packageGroup.convention(extension.packageGroup)
+            }
             val findStagingRepository = rootProject.tasks.register<FindStagingRepository>(
                 "find${capitalizedName}StagingRepository",
-                rootProject.objects,
-                extension,
-                repository,
                 registry
             )
             findStagingRepository {
-                description = "Finds the staging repository for ${repository.name}"
+                this.group = PublishingPlugin.PUBLISH_TASK_GROUP
+                this.description = "Finds the staging repository for ${repository.name}"
+                this.repository.convention(repository)
+                this.packageGroup.convention(extension.packageGroup)
+                this.descriptionRegex.convention(extension.repositoryDescription.map { "\\b" + Regex.escape(it) + "(\\s|$)" })
```

Comment:
@szpak These `this.` prefixes might look strange, but without it, future evolutions might break things, because there's a repository (from .all) and a task (from register/invoke) in scope. Alternatively this big setup block could be extracted to a separate function with repository as the parameter not a receiver:
```
        extension.repositories.all {
            extracted(this, rootProject, extension, registry)
        }
```

Edit:I did this in 777dbdeb